include/seastar/core/condition-variable.hh

```diff
@@ -76,6 +76,10 @@ private:
     // the base for queue waiters. looks complicated, but this is
     // to make it transparent once we add non-promise based nodes
     struct waiter : public boost::intrusive::list_base_hook<boost::intrusive::link_mode<boost::intrusive::auto_unlink>> {
+        waiter() = default;
+        waiter(waiter&&) = default;
+        waiter(const waiter&) = delete;
+        waiter& operator=(const waiter&) = delete;
```

Comment:
Sorry for being very picky, but does definition of the move constructor (even being =default one) make compiler delete the copy- one on its own?

Reply:
i think so. i am fine removing this. i made this very change based on your comment of 

> Maybe the copy constructor should be explicitly deleted instead?